### PR TITLE
Akka-http-client support

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR is for replacing dispatch with akka-http-client. The basics of the library have been migrated, however its still not compiling because I am holding off until #13 is decided

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -2,8 +2,8 @@ stripe-scala {
   endpoint="https://api.stripe.com"
   endpoint=${?STRIPE_ENDPOINT}
 
-  file-upload-endpoint="https://uploads.stripe.com"
-  file-upload-endpoint=${?STRIPE_FILE_UPLOAD_ENDPOINT}
+  fileUploadEndpoint="https://uploads.stripe.com"
```
Comment (leonardehrenfried) [inline src/main/resources/application.conf]: I was under the impression that the convention Typesafe config was to use hyphens:

See the following examples:
- https://github.com/typesafehub/config
- https://github.com/akka/akka/blob/master/akka-remote/src/main/resources/reference.conf

Comment (mdedetrich) [reply src/main/resources/application.conf]: Okay my bad, had no idea there were conventions. I will change the config to reflect this

**Comment by mdedetrich** [comment]
@leonardehrenfried Do you want to quickly review this? All tests are passing

**Comment by leonardehrenfried** [comment]
Okay, will take a look at the weekend.

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -5,8 +5,10 @@ stripe-scala {
...
 
-  apiKey = ${STRIPE_API_KEY}
+  api-key = ${STRIPE_API_KEY}
 
-  numberOfRetries=5
-  numberOfRetries=${?STRIPE_NUMBER_OF_RETRIES}
+  number-of-retries=5
+  number-of-retries=${?STRIPE_NUMBER_OF_RETRIES}
+
+  file-upload-chunk-timeout="5 minutes"
```
Comment (leonardehrenfried) [inline src/main/resources/application.conf:13]: Is 5 minutes a good default timeout?

**Review by leonardehrenfried — CHANGES_REQUESTED** [review]
Whilst I think dispatch is good enough I wouldn't be against moving to akka-http. 

However, I think that the users of the library shouldn't have to know what these things are:

```scala
implicit val system       = ActorSystem()
implicit val materializer = ActorMaterializer()
implicit val http         = Http()
```

Would there be a way to hide these from the API surface and provide default implementations instead? Maybe it could be overridden if someone really cares but I believe the default should be to use the library-provided actor systems and materializers.

File: src/main/resources/application.conf (hunk 2 above)
Comment (mdedetrich) [reply src/main/resources/application.conf:13]: No idea, theoretically you should never hit this timeout, its only the timeout for reading the input stream. Do you have a better idea for a timeout? Else I was thinking of leaving it and if it causes a problem I can always change it later

**Comment by mdedetrich** [comment]
> Would there be a way to hide these from the API surface and provide default implementations instead? Maybe it could be overridden if someone really cares but I believe the default should be to use the library-provided actor systems and materializers.

I can combine them and then use them as default parameters or I can provide them in the `org.mdedetrich.stripe.Config` object. Typically these are implicits which you would place once in an object and then just use an import to provide them throughout your application.

I am more leaning towards documenting this, as its not hard to understand and its pretty much something that everyone has to do when working with other libraries

Note that in Dispatch's case, it basically provided its own singleton for `Http`, however making singletons in Scala isn't hard (just cache them as `lazy val`). Also currently you do have to provide an execution context (which needs to be provided implicitly for similar reasons)

File: src/main/resources/application.conf (hunk 2 above)
Comment (leonardehrenfried) [reply src/main/resources/application.conf:13]: Okay, good point.

**Comment by leonardehrenfried** [comment]
I think I would be more comfortable if the materializer, the actor system and the Http object would be contained in a single trait/object so the method signature is less cluttered. Also, instantiating this one extra object in order to use the library isn't too much to ask.

If this is addressed, 👍  from me.

**Comment by mdedetrich** [comment]
> I think I would be more comfortable if the materializer, the actor system and the Http object would be contained in a single trait/object so the method signature is less cluttered. Also, instantiating this one extra object in order to use the library isn't too much to ask.

Is it okay if I do this in a seperate PR as there already is an issue for this, i.e. https://github.com/mdedetrich/stripe-scala/issues/13 ? Otherwise I can do it now, but I am want to get this merged to 0.2.x as soon as I can since I have other work that I want to target 0.2.x

**Comment by leonardehrenfried** [comment]
Sure, of course.

**Pull request merged**